library/java/src/io/envoyproxy/envoymobile/engine/EnvoyEngineImpl.java

```diff
@@ -6,7 +6,10 @@
 
   private final long engineHandle;
 
-  public EnvoyEngineImpl() { this.engineHandle = JniLibrary.initEngine(); }
+  public EnvoyEngineImpl() {
+    System.loadLibrary("envoy_jni");
```

Comment:
Moved here since we need the library here. The only odd thing is that we also have native bindings in the `AndroidEngineImpl` class -- maybe we can put this load functionality into the `JniLibrary`?